src/pages/index.js

```diff
@@ -18,81 +19,223 @@ import {
+  "Deleting..."
+);
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card, cardID) {
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.showLoading();
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.hideLoading();
+      });
+  });
+
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.showLoading();
+
+    const avatarData = {
+      avatar: inputValues.imageURL,
+    };
+
+    api
+      .editProfileImage(avatarData)
+      .then(() => {
+        userinfoComponent.setUserImage(inputValues.imageURL);
+        editAvatarPopup.close();
+      })
+      .catch((err) => {
+        console.error("Error updating avatar:", err);
+      })
+      .finally(() => {
+        editAvatarPopup.hideLoading();
+      });
   },
-  containerSelector
+  "Saving..."
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+// Likes///
+function handleLikeClick(card) {
+  const isLiked = card.isLiked();
+
+  if (isLiked) {
+    api
+      .likeCountRemove(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
+
+const section = new Section({}, containerSelector);
 
 // Functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    (cardID) => handleDeleteClick(card, cardID),
+    handleLikeClick
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
-  addCardFormPopup.close();
+  // Show loading state
+  addNewCardButton.classList.add("loading");
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      const cardElement = renderCard(newCard);
+      section.addItem(cardElement);
+      addCardFormPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      // Hide loading state whether the operation succeeds or fails
+      addNewCardButton.classList.remove("loading");
+    });
 }
+const api = new Api(apiURL);
+
```

Comment:
Needs correction: Don't forget to update user info on the server when you submit the profile form